src/clojure/nrepl/server.clj

```diff
@@ -107,8 +132,9 @@
    * :port — defaults to 0, which autoselects an open port
    * :bind — bind address, by default \"::\" (falling back to \"localhost\"
        if \"::\" isn't resolved by the underlying network stack)
+   * :auth-wrapper - defaults to the result of `(default-auth-wrapper)`
```

Comment:
Sounds reasonable -- I can investigate.  If it's not straightforward, is that something you'd want in the first iteration, or is it possible that could be added later?  (I agree that if it's feasible, it'd be better to have it now.)

Reply:
If i pursue it, what were you contemplating here?  I could see doing something like what's done for `--handler`/`--middleware` (depending on how we resolve the middleware-related questions), but that still leaves the token.  I assume we'll need to read it from an environment variable, file, stdin, or similar so that it's not visible to others on the system (via `ps -au`, etc.), which might suggest adding a custom auth handler to cmdline.

I suppose we could also consider defining a "standard", e.g. to look for `~/.nrepl-auth-token` (or whatever) for a candidate token, as a default.

Reply:
For this type of authentication yes. On the server side everything's easy with respect to auth token and auth schemes. The real question is how to teach clients about different schemes. Guess the idea is to just hardcode the support for whatever auth schemes are known in the various clients.